### ISSUE TITLE
Add the various different plots and helper functions for metrics relevant for creating a budget

### ISSUE THREAD

**Issue by AKuederle** [learn-and-code/MoneyPy#12]
This is a "main" issue. This means there will likely be various smaller issues related to this one. Please link them back here.

**Comment by AKuederle** [comment]
Hey  when you started your discussion, involve me a little bit. This part is very much about the choice of tools. So it might be good if you have someone who can give you some inside experience.

**Comment by tweakimp** [comment]
I dont think I think of this issue in the same way as you do. I thought we are just thinking of functions to work with the data given. Like a function that does: 1) Get all money spent from database. 2)Plot it over time. Why do the tools matter for creating functions?

**Comment by AKuederle** [comment]
Sorry when I wasn't precise here. There are a wide variety of plotting tools for Python available (with tools I meant Python packages) and why it might be nice to support multiple of them, we should select one of them for now

**Comment by jxhdev** [comment]
I have very briefly used matplotlib in a tutorial but I'm down to check out other libraries too. 
@tweakimp I think that we can still write our own functions that calls the database for step 1) but for step 2) it would be a lot easier to do with the various data visualization libraries out there. I wouldn't even know how and where to start with writing our own data visualization functions.. But I'm up for it if we're feeling ambitious

**Comment by tweakimp** [comment]
Is it really that complicated? I thought we get a matrix containing all the data and we jsut write functions to plot parts of the data?

**Comment by AKuederle** [comment]
@jxhdev I don't think @tweakimp was planning on building his own visualization library (or were your? :D). Even with the existing ones you can get very lowlevel if this is required

**Comment by AKuederle** [comment]
And no it is not that complicated. The "complicated" part will be to decide on a good userinterface for the functions. Namely to find the balance between great defaults and customization. Making the main plots with any library would be pretty straight forward, I guess (depends of course on the polts you select)

**Comment by tweakimp** [comment]
getting everything to work should be the only priority here. we can still polish after the whole thing works

**Comment by BlitzKraft** [comment]
Since there are many choices of libraries, polishing later would involve a lot of redoing stuff. If we plan right now, it will reduce the amount of rework needed in the future.

**Comment by tweakimp** [comment]
I dont think we are talking about the same thing. I was thinking in this issue we discuss what data to present and how to process it, not how to actually visualize it. Once I have a matrix of the data I want to plot, it is really necessary to know how I want to plot it?

**Comment by BlitzKraft** [comment]
Not "necessary", but an agreement on the choice of tools by the team would be appreciated. matplotlib gets my vote, but if there are better tools suited here, I will be definitely reconsider.

**Comment by tweakimp** [comment]
This vid gives a nice overview. https://www.youtube.com/watch?v=FytuB8nFHPQ

**Comment by AKuederle** [comment]
My vote is on plotly or Bokeh for this project (in particular plotly). They can both be integrated nicely into a web interface (if we want something like this somewhere in the future), and are interactive by nature.

@tweakimp the actual visualization is the main part. Getting the data will hopefully just be one line if we build a proper ORM system in #11

**Comment by DanielPDWalker** [comment]
Made a decent start on the visualization in #25. Bit more to do, a color function in utilities for a start. Thinking also about more plots, anyone has suggestions for more? Currently there is a Grouped Bar Plot, a Line Plot and a Pie Plot.
